MekHQ/src/mekhq/campaign/personnel/autoAwards/AutoAwardsController.java

```diff
@@ -346,296 +346,288 @@ private void buildAwardLists(int awardListCase) {
-                                            scenarioAwards.add(award);
-                                        } else {
-                                            ignoredAwards.add(award);
-                                        }
-                                        break;
-                                    case "skill":
-                                        if (campaign.getCampaignOptions().isEnableSkillAwards()) {
-                                            skillAwards.add(award);
-                                        } else {
-                                            ignoredAwards.add(award);
-                                        }
-                                        break;
-                                    case "time":
-                                        if (campaign.getCampaignOptions().isEnableTimeAwards()) {
-                                            timeAwards.add(award);
-                                        } else {
-                                            ignoredAwards.add(award);
-                                        }
-                                        break;
-                                    case "training":
-                                        trainingAwards.add(award);
-                                        break;
-                                    default:
-                                        // if autoAwards doesn't know what to do with an Award, it ignores it
-                                        ignoredAwards.add(award);
-                                }
-                            }
-                            // These logs help users double-check that the number of awards found matches their records
-                            logger.info("autoAwards found {} Kill Awards (excluding Mission & Scenario Kill Awards)", killAwards.size());
-                            logger.info("autoAwards found {} Rank Awards", rankAwards.size());
-                            logger.info("autoAwards found {} Scenario Awards", scenarioAwards.size());
-                            logger.info("autoAwards found {} Skill Awards", skillAwards.size());
-                            logger.info("autoAwards found {} Time Awards", timeAwards.size());
-                            logger.info("autoAwards found {} Training Awards", trainingAwards.size());
-                            logger.info("autoAwards found {} Misc Awards", miscAwards.size());
-                            logger.info("autoAwards is ignoring {} Awards", ignoredAwards.size());
+        for (String setName : allSetNames) {
+            if (Arrays.asList(filterList).contains(setName)) {
+                logger.info("'{}' was found in the list of sets to be ignored. Ignoring it.", setName);
+                continue;
+            }
 
+            logger.info("Getting all awards from set: {}", setName);
+
+            awards.addAll(AwardsFactory.getInstance().getAllAwardsForSet(setName));
+        }
```

Comment:
This PR looks like it's has a lot of changes, but pretty much all I did was close this loop properly, so the below switches weren't included and cleaned things up a bit.

One day I would love to go through and rewrite autoAwards with all the cool tricks I now know.